Src/Newtonsoft.Json/Utilities/TypeExtensions.cs

```diff
@@ -593,27 +617,16 @@ public static bool AssignableToTypeName(this Type type, string fullTypeName, boo
                 current = current.BaseType();
             }
 
-            if (searchInterfaces)
-            {
-                foreach (Type i in type.GetInterfaces())
-                {
-                    if (string.Equals(i.Name, fullTypeName, StringComparison.Ordinal))
-                    {
-                        match = type;
-                        return true;
-                    }
-                }
-            }
```

Comment:
Did you remove this because it caused trouble and `searchInterfaces` was always false?